backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java

```diff
@@ -871,6 +1060,19 @@ public void setProjectShortName(
       var project =
           projectDAO.setProjectShortName(request.getId(), request.getShortName(), userInfo);
       var response = SetProjectShortName.Response.newBuilder().setProject(project).build();
+
+      // Add succeeded event in local DB
+      JsonObject eventMetadata = new JsonObject();
+      eventMetadata.addProperty("entity_id", project.getId());
+      eventMetadata.addProperty("updated_field", "short_name");
+      eventMetadata.addProperty("updated_field_value", request.getShortName());
+      eventMetadata.addProperty("message", "project short_name updated successfully");
+      futureEventDAO.addLocalEventWithBlocking(
+          ModelDBServiceResourceTypes.PROJECT.name(),
+          "update.resource.project.update_project_succeeded",
+          project.getWorkspaceServiceId(),
+          eventMetadata);
```

Comment:
Do you know what is the right format for the eventMetadata field? According to https://github.com/VertaAI/protos-all/blob/master/protos/protos/private/uac/Event.proto#L15 it should be JSON consumable by external service. I'm not sure that format that you are using here is the right one. Could you at least extract the format to a separate method to easily change it later?

Reply:
But can you add a common method for projects?

Reply:
I agree with Evgenii that there should be a common method for projects. It would need to take the updated field type, the updated field/s (attributes, tags, etc.), and the message, but then the method could handle putting the JSON together and calling the event service.